src/modules/transaction/utils.ts

```diff
@@ -224,3 +300,21 @@ export function* waitForTx(txHash: string) {
     }
   }
 }
+
+export const takeEverySuccessfulTx = (
+  actionType: string,
+  worker: (...args: any[]) => any,
+  ...args: Parameters<(...args: any[]) => any>
+) =>
+  fork(function*() {
+    while (true) {
+      const action: FetchTransactionSuccessAction = yield take([
+        FETCH_TRANSACTION_SUCCESS,
+        FETCH_CROSS_CHAIN_TRANSACTION_SUCCESS
+      ])
+
+      if (action.payload.transaction.actionType === actionType) {
```

Comment:
not fully sure I undestand how this works, the `actionType` will be `FETCH_TRANSACTION_SUCCESS` or `FETCH_CROSS_CHAIN_TRANSACTION_SUCCESS` right? (the string values from those consts) but in the marketplace is called with `BUY_ITEM_CROSS_CHAIN_SUCCESS`. Will they match?

Reply:
The transaction has the `actionType` which is the original action that triggered the fetch transaction action. We can track them this way and act upon it